posts/formalizing-isocrystals-in-lean.md

```diff
@@ -0,0 +1,211 @@
+---
+author: 'Robert Y. Lewis, Heather Macbeth'
+category: 'New in mathlib'
+date: 2022-07-19 19:30:08 UTC+02:00
+description: ''
+has_math: true
+link: ''
+slug: classification-of-one-dimensional-isocrystals
+tags: ''
+title: Classification of one-dimensional isocrystals
+type: text
+---
+
+Last year, there was a [big mathlib refactor](https://leanprover-community.github.io/blog/posts/semilinear-maps/) to replace linear maps throughout the library with  *semilinear maps*,
+a more abstract concept which, importantly, unifies linear and conjugate-linear maps.
+
+But this is not the full extent of the generalization!  Our number theorist friends here in mathlib told us that we should
+make sure we chose this full generality of *semilinear* maps, maps $f:M \to N$ such that $f(ax)=\sigma(a)f(x)$ for some ring homomorphism $\sigma$ between the scalar rings of the modules $M$ and $N$.  So we and our coauthor Frédéric Dupuis implemented this full generality, and then asked them for an example to illustrate their need for this more abstract definition. This blog post tells the story of our little adventure in number theory.
+
+It turns out that the standard use of semilinear maps in number theory is for *Frobenius-semilinearity*, semilinearity with
+respect to the ring homomorphism of the fraction field of the $p$-typical Witt vectors over a perfect characteristic-$p$ integral domain which is induced by the Frobenius
+automorphism of that domain.  Let's backtrack to catch everyone up...
+
+<!-- TEASER_END -->
+
+# Witt vectors and isocrystals
+
+Back in 2020, one of us (Rob), together with Johan Commelin, [formalized the theory of Witt vectors](https://arxiv.org/abs/2010.02595) in Lean.  The $p$-typical Witt 
+vectors $\mathbb{W}(R)$ over a ring $R$  are, concretely, sequences of elements of $R$, but equipped with a rather wild multiplicative and additive
+structure (dependent on $p$) to make this set into a commutative ring.  The canonical example is the $p$-adic numbers, which are the Witt vectors
+of $\mathbb{Z}/p\mathbb{Z}$.
+
+A ring $R$ of characteristic $p$ has an endomorphism, constructed by sending each element $x$ to $x ^ p$.  $R$ is called *perfect* if this
+endomorphism is an automorphism.  And under mild further conditions ($R$ an integral domain) this automorphism "lifts" <!-- check terminology -->
+to an automorphism of the field of fractions of $\mathbb{W}(R)$, which we will denote $\varphi$.  It's *Frobenius-semilinearity*, i.e. semilinearity with respect $\varphi$, that the number theorists
+want to consider! 
+
+The fundamental result here is a classification theorem for
+Frobenius-semilinear automorphisms of finite-dimensional vector spaces over $K$, the field of fractions of $\mathbb{W}(k)$, for $k$ an algebraically closed field (and hence a perfect integral domain).  Such an object (a finite-dimensional 
```

Comment:
Maybe introduce `K` as notation for the field of fractions in the preceding paragraph, when you also introduce `ϕ`?

Reply:
I think it can't quite be done while remaining literally correct, because in the previous paragraph we discuss the fraction field of `𝕎 R`, but in this paragraph we discuss the fraction field of `𝕎 k`, for `k` some algebraically closed field.  So to remain literally correct one would have to refer to `K(R)`, `K(k)`, etc. which seems unwieldy.